test/jdk/java/nio/file/Files/InputStreamTest.java

```diff
@@ -123,6 +128,17 @@ static void testSkip(Path tmpdir) throws IOException {
         }
     }
 
+    /**
+     * Tests that Files.newInputStream(Path).available() does not throw
+     */
+    static void testAvailable() throws IOException {
+        Path stdin = Path.of("/dev", "stdin");
+        if (Files.exists(stdin)) {
+            InputStream s = Files.newInputStream(stdin);
+            s.available();
+        }
```

Comment:
I assume you meant to close the input stream.

What about the other methods defined by InputStream that have special handling in ChannelInputStream? I assume we should add test coverage for these methods.

Reply:
I added test coverage for `skip` but not for `readAllBytes` nor `readNBytes`:
- `readAllBytes` throws before but hangs after the change;
- `readNBytes` hangs both before and after the change.

Reply:
I wonder if there is another special file that could be read to EOF to allow the methods be tested. Asking because the overrides have a code path we need to test.

Reply:
I need to investigate this. I might hold off integrating even thought it's approved until this is determined.

Reply:
> I wonder if there is another special file that could be read to EOF to allow the methods be tested.

I think that this can done using a named pipe, at least on UNIX.